nucypher/cli/characters/alice.py

```diff
@@ -7,44 +7,201 @@
 from nucypher.config.characters import AliceConfiguration
 from nucypher.config.keyring import NucypherKeyring
+from nucypher.utilities.sandbox.constants import TEMPORARY_DOMAIN
 
 
-# Args (geth, provider_uri, federated_only, dev, pay_with, network, registry_filepath)
-from nucypher.utilities.sandbox.constants import TEMPORARY_DOMAIN
+option_bob_verifying_key = click.option(
+    '--bob-verifying-key', help="Bob's verifying key as a hexadecimal string", type=click.STRING,
+    required=True)
+option_pay_with = click.option(
+    '--pay-with', help="Run with a specified account", type=EIP55_CHECKSUM_ADDRESS)
+
+
+class AliceConfigOptions:
+
+    __option_name__ = 'config_options'
+
+    def __init__(
+            self, dev, network, provider_uri, geth, federated_only, discovery_port,
+            pay_with, registry_filepath, middleware):
 
+        if federated_only and geth:
+            raise click.BadOptionUsage(
+                option_name="--geth",
+                message="--federated-only cannot be used with the --geth flag")
 
-def _admin_options(func):
-    @click.option('--geth', '-G', help="Run using the built-in geth node", is_flag=True)
-    @click.option('--provider', 'provider_uri', help="Blockchain provider's URI", type=click.STRING)
-    @click.option('--federated-only', '-F', help="Connect only to federated nodes", is_flag=True)
-    @click.option('--dev', '-d', help="Enable development mode", is_flag=True)
-    @click.option('--pay-with', help="Run with a specified account", type=EIP55_CHECKSUM_ADDRESS)
-    @click.option('--network', help="Network Domain Name", type=click.STRING)
-    @click.option('--registry-filepath', help="Custom contract registry filepath", type=EXISTING_READABLE_FILE)
-    @functools.wraps(func)
-    def wrapper(*args, **kwargs):
-        return func(*args, **kwargs)
-    return wrapper
-
-
-# Args (geth, provider_uri, federated_only, dev, pay_with, network, registry_filepath, config_file, discovery_port,
-#       hw_wallet, teacher_uri, min_stake)
-def _api_options(func):
-    @_admin_options
-    @click.option('--config-file', help="Path to configuration file", type=EXISTING_READABLE_FILE)
-    @click.option('--discovery-port', help="The host port to run node discovery services on", type=NETWORK_PORT)
-    @click.option('--hw-wallet/--no-hw-wallet', default=False)
-    @click.option('--teacher', 'teacher_uri', help="An Ursula URI to start learning from (seednode)", type=click.STRING)
-    @click.option('--min-stake', help="The minimum stake the teacher must have to be a teacher", type=click.INT,
-                  default=0)
-    @functools.wraps(func)
-    def wrapper(*args, **kwargs):
-        return func(*args, **kwargs)
-    return wrapper
+        # Managed Ethereum Client
+        eth_node = NO_BLOCKCHAIN_CONNECTION
+        if geth:
+            eth_node = actions.get_provider_process()
+            provider_uri = ETH_NODE.provider_uri(scheme='file')
```

Comment:
@fjarri - I was working on top of this branch and noticed the that `ETH_NODE` is not defined - it should be `eth_node`.